src/app/proposals/components/OZGovExecute.tsx

```diff
@@ -0,0 +1,128 @@
+import { Proposal } from "@/app/api/common/proposals/proposal";
+import Tenant from "@/lib/tenant/tenant";
+import {
+  useAccount,
+  useContractRead,
+  useContractWrite,
+  useWaitForTransaction,
+} from "wagmi";
+import { Button } from "@/components/ui/button";
+import { proposalToCallArgs } from "@/lib/proposalUtils";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { blocksToSeconds } from "@/lib/blockTimes";
+
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+
+interface Props {
+  proposal: Proposal;
+}
+
+export const OZGovExecute = ({ proposal }: Props) => {
+  const { contracts } = Tenant.current();
+  const { address } = useAccount();
+  const [canExecute, setCanExecute] = useState(false);
+  const [executeTime, setExecuteTime] = useState(new Date());
+
+  // Check whether user has the EXECUTOR_ROLE
+  const { data: hasExecuteRole } = useContractRead({
+    address: contracts.timelock!.address as `0x${string}`,
+    abi: contracts.timelock!.abi,
+    functionName: "hasRole",
+    args: [
+      "0xd8aa0f3194971a2a116679f7c2090f6939c8d4e01a2a8d7e41d55e5351469e63", // EXECUTOR_ROLE
```

Comment:
Is this the same for all OZ gov contracts? Maybe we could move this into a const

Reply:
I am not sure about the other contracts. ENS modified their version a bit. Good call on making this a constant though